src/unearth/auth.py

```diff
@@ -284,27 +284,26 @@ def _should_save_password_to_keyring(self) -> bool:
             return False
         return input("Save credentials to keyring [y/N]: ") == "y"
 
-    def handle_400s(self, resp: Response, **kwargs: Any) -> Response:
-        # We only care about 401, 403, 404 responses, anything else we want to just
+    def handle_401(self, resp: Response, **kwargs: Any) -> Response:
+        # We only care about 401 response, anything else we want to just
         #   pass through the actual response
-        if resp.status_code not in [401, 403, 404]:
+        if resp.status_code != 401:
             return resp
 
         parsed = urlparse(cast(str, resp.url))
 
         # Query the keyring for credentials:
         username, password = self._get_new_credentials(
             resp.url,
-            allow_netrc=True,
+            allow_netrc=False,
             allow_keyring=True,
         )
 
         # Prompt the user for a new username and password
         save = False
         if not username and not password:
             # We are not able to prompt the user so simply return the response
-            # Also, do not prompt on 404
-            if not self.prompting or resp.status_code == 404:
+            if not self.prompting:
```

Comment:
Don't prompt for either 403/404
```suggestion
            if not self.prompting or resp.status_code in [403,404]:
```